DolbyIO/Source/Private/DolbyIOSubsystem.cpp

```diff
@@ -208,6 +208,40 @@ void UDolbyIOSubsystem::Initialize(const FString& Token)
 			            BroadcastEvent(OnVideoTrackRemoved, VideoTrack);
 		            });
 	        })
+	    .then(
+	        [this](event_handler_id)
+	        {
+		        return Sdk->device_management().add_event_handler(
+		            [this](const audio_device_changed& Event)
+		            {
+			            auto ChangedDirection = ToEDolbyIODeviceDirection(Event.utilized_direction);
+			            if (!Event.device)
+			            {
+				            DLB_UE_LOG("Audio device changed to NO device for direction: %d", int(ChangedDirection));
+				            BroadcastEvent(OnCurrentAudioDeviceChanged, true, FDolbyIOAudioDevice{}, ChangedDirection);
+				            return;
+			            }
+			            Sdk->device_management()
+			                .get_audio_devices()
+			                .then(
+			                    [this, ChangedDirection,
+			                     SdkDeviceId = *Event.device](const std::vector<audio_device>& DvcDevices)
+			                    {
+				                    for (const audio_device& Device : DvcDevices)
+					                    if (SdkDeviceId == Device.get_identity())
+					                    {
+						                    DLB_UE_LOG("Audio device CHANGED for direction: %d to device - name: %s, "
```

Comment:
CHANGED looks like you're shouting at the user :P Also I would prefer logging directions as strings instead of casting them to numbers.

Reply:
Actually it makes easier to me read logs ;)
I will lower it for others. ;)